src/Files.App/Actions/Content/ImageManipulation/BaseRotateAction.cs

```diff
@@ -20,8 +21,14 @@ internal abstract class BaseRotateAction : ObservableObject, IAction
 		protected abstract BitmapRotation Rotation { get; }
 
 		public bool IsExecutable =>
-			IsContextPageTypeAdaptedToCommand() &&
-			(context.ShellPage?.SlimContentPage?.SelectedItemsPropertiesViewModel?.IsCompatibleToSetAsWindowsWallpaper ?? false);
+			context.ShellPage is not null &&
+			context.ShellPage.SlimContentPage is not null &&
+			context.PageType != ContentPageTypes.RecycleBin &&
+			context.PageType != ContentPageTypes.ZipFolder &&
+			context.PageType != ContentPageTypes.ReleaseNotes &&
+			context.PageType != ContentPageTypes.Settings &&
```

Comment:
If you're interested, you can use is, and, or:

```suggestion
			context.PageType is not ContentPageTypes.RecycleBin and
			not ContentPageTypes.ZipFolder and
			not ContentPageTypes.ReleaseNotes and
			not ContentPageTypes.Settings &&
```